jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt

```diff
@@ -10,13 +10,14 @@ import com.intellij.openapi.editor.event.EditorFactoryListener
 import com.intellij.openapi.editor.impl.EditorImpl
 import com.intellij.psi.PsiDocumentManager
 import software.aws.toolkits.jetbrains.services.codewhisperer.editor.CodeWhispererEditorUtil.codeWhispererLanguage
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhispererExplorerActionManager
 import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus
 import software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererCodeCoverageTracker
 
 class CodeWhispererEditorListener : EditorFactoryListener {
     override fun editorCreated(event: EditorFactoryEvent) {
         val editor = (event.editor as? EditorImpl) ?: return
-
+        if (!CodeWhispererExplorerActionManager.getInstance().hasAcceptedTermsOfService()) return
```

Comment:
No I don't mean this one, I mean to put this check only inside the start of the below documentChanged() event. Since doing this will miss for the cases when the users have the editor opened when CodeWhisperer was not enabled yet, but then enabled after. This event is only called when the editor is initially created, thus for all these editors they will miss CodeWhisperer logic. By putting it below we are saying "we are adding this documentListener to the editor anyway but do nothing if CWSPR is not enabled"

Reply:
ack make sense